generator/README.md

```diff
@@ -0,0 +1,831 @@
+
+## Requirements and Dependencies
+
+The script requires **Python 3** pre-installed on the host system. The minimal supported Python 3 version is **3.7.6**. It may work on versions back to 3.5 (the minimal version that has not yet reached [the end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)), but this is not supported and may break in the future.
+
+Note: To install the dependencies for this script, you must use the **pip3** command.
+
+All required libraries are listed in `requirements.txt` and should be pre-installed on the system prior to using the sript. Please use the following command to install the libraries:
+
+```shell script
+$ pip3 install -r generator/requirements.txt
+```
+
+Please also make sure all git submodules are installed and up to date since the script uses the XML parser provided in a submodule.
+
+```shell script
+$ git submodule update --init --recursive
+```
+
+## Usage
+
+**Usage example**
+
+```shell script
+$ cd sdl_ios
+$ python3 generator/generator.py -xml generator/rpc_spec/MOBILE_API.xml -xsd generator/rpc_spec/MOBILE_API.xsd -d output_dir
+```
+
+
+As a result the output_dir will have all the new generated files.
+
+**Detailed usage description (keys, options)**
+
+```
+usage: generator.py [-h] [-v] [-xml SOURCE_XML] [-xsd SOURCE_XSD]
+                    [-d OUTPUT_DIRECTORY] [-t [TEMPLATES_DIRECTORY]]
+                    [-r REGEX_PATTERN] [--verbose] [-e] [-s] [-m] [-y] [-n]
+
+Proxy Library RPC Generator
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -v, --version         print the version and exit
+  -xml SOURCE_XML, --source-xml SOURCE_XML, --input-file SOURCE_XML
+                        should point to MOBILE_API.xml
+  -xsd SOURCE_XSD, --source-xsd SOURCE_XSD
+  -d OUTPUT_DIRECTORY, --output-directory OUTPUT_DIRECTORY
+                        define the place where the generated output should be
+                        placed
+  -t [TEMPLATES_DIRECTORY], --templates-directory [TEMPLATES_DIRECTORY]
+                        path to directory with templates
+  -r REGEX_PATTERN, --regex-pattern REGEX_PATTERN
+                        only elements matched with defined regex pattern will
+                        be parsed and generated
+  --verbose             display additional details like logs etc
+  -e, --enums           if present, all enums will be generated
+  -s, --structs         if present, all structs will be generated
+  -m, -f, --functions   if present, all functions will be generated
+  -y, --overwrite       force overwriting of existing files in output
+                        directory, ignore confirmation message
+  -n, --skip            skip overwriting of existing files in output
+                        directory, ignore confirmation message
+```
+
+### How to use the generated classes
```

Comment:
Can you add a section here on how to run the unit tests?

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/a179e65851c4c239d5bef67a0f3ca591dd235dec